src/backend/aspen/fileio/fasta_streamer.py

```diff
@@ -52,8 +68,37 @@ def stream(self):
                     in self.cansee_groups_private_identifiers
                     or self.user.system_admin
                 ):
-                    yield (f">{sample.private_identifier}\n")  # type: ignore
+                    yield self._output_id_line(sample.private_identifier)
                 else:
-                    yield (f">{sample.public_identifier}\n")
-                yield (stripped_sequence)
-                yield ("\n")
+                    yield self._output_id_line(sample.public_identifier)
+                yield stripped_sequence
+                yield "\n"
+
+    def _output_id_line(self, identifier) -> str:
+        """Produces the ID line for current sequence in fasta.
+
+        Certain downstream consumers (eg, UShER) restrict what characters can be
+        used in the ID. Also handles any modifications that must be made to ID
+        characters so they don't break the downstream consumer."""
+        if self.downstream_consumer == SpecialtyDownstreams.USHER.value:
+            output_ready_id = self._handle_usher_id(identifier)
+        else:
+            output_ready_id = identifier
+        return f">{output_ready_id}\n"
+
+    def _handle_usher_id(self, identifier) -> str:
```

Comment:
Also nit: Ideally I'd prefer to break these filters out into their own classes and pass them into the FastaStreamer constructor. but since we only have one downstream atm, we can stress this later 😄